### PR TITLE
Add Pinned Recording API endpoint

### PR DESCRIPTION
**Pull request by jdaok**
TO DO:
- [x] Add tests

This PR aims to add the Pinned Recording API endpoint which contains routes to 

- pin, 
https://api.listenbrainz.org/1/pin
- unpin, 
https://api.listenbrainz.org/1/pin/unpin
- delete pin, 
https://api.listenbrainz.org/1/pin/delete/(pinned-id)
- get pin history for a user, 
https://api.listenbrainz.org/1/(user-name)/pins
- and get pin for a user's following list.
https://api.listenbrainz.org/1/(user-name)/pins/following

### CODE REVIEW

**Review by mayhem — CHANGES_REQUESTED** [review]
I've skimmed this and I have a couple of requests:

1. Change the base branch to be the branch from #1510 -- which makes this PR easier to read.
2. In the description of the PR, please add a list of the URLs for the endpoints you're adding. That will help us review them to ensure they are consistent and RESTy (e.g. some have the word get in the URL, which is not RESTy).

**Comment by jdaok** [comment]
@mayhem Added URLs for the added endpoints in the description. As lucifer said I can't update the base branch but [1510](https://github.com/metabrainz/listenbrainz-server/pull/1510) is done and I can merge it's changes here as soon as it's merged into master.

**Comment by mayhem** [comment]
Ok, after a chat with alastair, we think the latter two endpoint URLs should be:

1. /1/{username}/pins
2. /1/{username}/pins/following

This makes these endpoint consistent with other interfaces, such as playlists. Go ahead, adjust these and then add tests.

**Comment by jdaok** [comment]
@mayhem All done :) The new tests are failing because of the feature flag but they pass on my local machine.

File: listenbrainz/webserver/__init__.py [hunk 1]
```diff
@@ -309,3 +309,7 @@ def _register_blueprints(app):
 
     from listenbrainz.webserver.views.user_timeline_event_api import user_timeline_event_api_bp
     app.register_blueprint(user_timeline_event_api_bp, url_prefix=API_PREFIX)
+
+    if app.config.get("FEATURE_PINNED_REC", False):
```
Comment (amCap1712) [inline listenbrainz/webserver/__init__.py:313]: You also need to add the feature flag to this file, https://github.com/metabrainz/listenbrainz-server/blob/master/consul_config.py.ctmpl . I'll update the production config and let you know the exact changes needed to this file.

Comment (jdaok) [reply listenbrainz/webserver/__init__.py:313]: I added FEATURE_PINNED_REC = '''{{template "KEY" "feature_pinned_rec"}}'''  following the pattern from other commits, please let me know if this is sufficient.

File: listenbrainz/tests/integration/test_pinned_recording_api.py [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+        self.assertEqual(response.json["code"], 404)
+
+    def test_get_pins_for_user(self):
+        """Test that valid response is received with 200 code"""
+        count = self.insert_test_data(self.user["id"], 2)  # pin 2 recordings
+        response = self.client.get(url_for("pinned_rec_api_bp_v1.get_pins_for_user", user_name=self.user["musicbrainz_id"]))
+        self.assert200(response)
+        data = json.loads(response.data)
+
+        # print(json.dumps(data, indent=2))
```
Comment (mayhem) [inline listenbrainz/tests/integration/test_pinned_recording_api.py:226]: remove.

File: listenbrainz/tests/integration/test_pinned_recording_api.py [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+        pin1 = self.pin_single_sample(self.followed_user_1["id"], 0)  # pin recording for followed_user_1
+        pin2 = self.pin_single_sample(self.followed_user_2["id"], 1)  # pin recording for followed_user_2
+
+        response = self.client.get(
+            url_for("pinned_rec_api_bp_v1.get_pins_for_user_following", user_name=self.user["musicbrainz_id"])
+        )
+        self.assert200(response)
+        data = json.loads(response.data)
+
+        print(json.dumps(data, indent=2))
```
Comment (mayhem) [inline listenbrainz/tests/integration/test_pinned_recording_api.py:356]: nuke.

File: listenbrainz/tests/integration/test_pinned_recording_api.py [hunk 4]
```diff
@@ -0,0 +1,470 @@
...
+        self.pin_single_sample(self.followed_user_1["id"], 0)  # pin recording for followed_user_1
+        included_pin = self.pin_single_sample(self.followed_user_2["id"], 1)  # pin recording for followed_user_2
+
+        limit = 1
+        response = self.client.get(
+            url_for("pinned_rec_api_bp_v1.get_pins_for_user_following", user_name=self.user["musicbrainz_id"], count=limit)
+        )
+        data = json.loads(response.data)
+        pins = data["pinned_recordings"]
+        print(json.dumps(data, indent=2))
```
Comment (mayhem) [inline listenbrainz/tests/integration/test_pinned_recording_api.py:395]: 'aight, scan the module for any other prints. :)

**Review by mayhem — APPROVED** [review]
Aside from tiny bits of cleanup, I'm happy!

File: listenbrainz/tests/integration/test_pinned_recording_api.py (hunk 4 above)
Comment (jdaok) [reply listenbrainz/tests/integration/test_pinned_recording_api.py]: Oops. Done

File: listenbrainz/webserver/__init__.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz/webserver/__init__.py:313]: I updated the consul configs. Can you please merge the latest master into your branch and then change this line to `FEATURE_PINNED_REC = {{template "KEY_JSON" "feature_pinned_rec"}}`.

Comment (jdaok) [reply listenbrainz/webserver/__init__.py:313]: Changed!

**Review by amCap1712 — APPROVED** [review]
I pushed one small commit to fix docs. LGTM and Awesome Work! 💯

**Pull request merged**